packages/pool/contracts/DelegationUtils.sol

```diff
@@ -78,9 +78,11 @@ abstract contract DelegationUtils is RewardUtils, IDelegationUtils {
         payReward();
         User storage user = users[msg.sender];
         address userDelegate = userDelegate(msg.sender);
+        require(userDelegate != address(0), ERROR_UNAUTHORIZED);
```

Comment:
We are checking this here because if an undelegated user (which appears to have delegated to `address(0)`) calls `undelegateVotingPower()`, it should revert (because undelegated users shouldn't be able to undelegate). What we check in another method isn't relevant because the user may call this one directly.